sql/core/src/main/scala/org/apache/spark/sql/execution/command/views.scala

```diff
@@ -125,6 +145,14 @@ case class CreateViewCommand(
         val viewIdent = tableMetadata.identifier
         checkCyclicViewReference(analyzedPlan, Seq(viewIdent), viewIdent)
 
+        // uncache the cached data before replacing an exists view
+        logInfo(s"Try to uncache ${viewIdent.quotedString} before replacing.")
+        try {
+          sparkSession.catalog.uncacheTable(viewIdent.quotedString)
+        } catch {
+          case NonFatal(e) => log.warn(e.toString, e)
+        }
```

Comment:
How about just moving this pattern (`try { uncacheTable(..) } catch { case NonFatal... } `) into `CommandUtils` then using the util func from the related commands, `create view`, `drop table`, ...?

Reply:
Done